### ISSUE TITLE
Can't even get started because of build issues

### ISSUE THREAD

**Issue by ksvanhorn** [ucsd-progsys/liquidhaskell-tutorial#105]
Following the tutorial's instructions, I tried to download and build the liquidhaskell-tutorial package using stack. It says to "iteratively edit/compile... until it builds", but gives no guidance on how to fix the errors. The error I'm getting is this:

    In the dependencies for text-format-0.3.2:
        base-4.14.0.0 from stack configuration does not match >=4.3 && <4.13  (latest
                      matching version is 4.12.0.0)
    needed due to liquidhaskell-tutorial-0.2.0.0 -> text-format-0.3.2

I don't know any good way to fix this.

**Comment by ranjitjhala** [comment]
How did you get this error message? That is, can you tell me the sequence
of steps to reproduce the error? (eg are you using cabal or stack?)

**Comment by ksvanhorn** [comment]
I used the stack command mentioned in the tutorial:

stack build --fast --file-watch

One odd thing I noticed. When I first was looking into this, I checked out the page for the text-format package on hackage.haskell.org <http://hackage.haskell.org/>, and it listed the dependency “base (>= 4.3 && <4.13)”, matching the error message I got. After getting your email reply I checked the page again, and it now says “base (>= 4.3 && <4.15)”. Nonetheless, running “stack build --fast --file-watch” still fails with the same error message.

**Comment by ranjitjhala** [comment]
Hmm. So you did:

```
$ git clone https://github.com/ucsd-progsys/liquidhaskell-tutorial.git
$ cd liquidhaskell-tutorial
$ stack build --fast --file-watch
```

and that tickles the error? Are you on Windows by any chance? Also, which
version of `stack` do you have?

**Comment by ksvanhorn** [comment]
Yes, that’s right. No, I’m not on Windows, I’m on OS X 10.15 (Catalina). I originally had stack version 2.3.3, and in the process of trying things to fix the problem upgraded to version 2.5.1 via “stack upgrade”.

More info: suspecting that for some reason stack had out-of-date information about text-format-0.3.2, I tried various things, including stack upgrade, and finally completely removed the liquidhaskell-tutorial directory and then ran those three commands (git, cd, stack) again. This time the build went fine until it came to liquid-ghc-prim -- at that point it informed me that I needed to install an SMC solver.

So it looks to me like maybe my guess about out-of-date package information was right.

At the current moment I’m trying to figure out how to install Z3. I downloaded the Mac OS binaries and unzipped, but I don’t know where to install the results so that stack/ghc can find it. Theres an include directory, and a bin directory with various static and dynamic libraries as well as a python subdirectory under that.

**Comment by ranjitjhala** [comment]
Great - just copy the z3 binary/executable to someplace that’s in your
PATH, that should do it.
